agent/fw_drupal8.c

```diff
@@ -605,6 +605,89 @@ NR_PHP_WRAPPER(nr_drupal94_invoke_all_with_clean) {
 NR_PHP_WRAPPER_END
 #endif  // OAPI
 
+/*
+ * Purpose: Instrument Drupal Attribute Hooks for Drupal 11.1+
+ *
+ * Params: 1. A zval pointer to the moduleHandler instance in use by Drupal.
+ *
+ * Return: bool
+ *
+ */
+static bool nr_drupal_hook_attribute_instrument(zval* module_handler) {
+  zval* hook_implementation_map = NULL;
+
+  nr_php_string_hash_key_t* hook_key = NULL;
+  zval* hook_val = NULL;
+  nr_php_string_hash_key_t* class_key = NULL;
+  zval* class_val = NULL;
+  nr_php_string_hash_key_t* method_key = NULL;
+  zval* module_val = NULL;
+
+  char* hookpath = NULL;
+
+  hook_implementation_map = nr_php_get_zval_object_property(
+      module_handler, "hookImplementationsMap");
+
+  if (NULL == hook_implementation_map) {
+    nrl_warning(NRL_FRAMEWORK, "NULL hookImplementationsMap object property");
+    return false;
+  }
+
+  if (!nr_php_is_zval_valid_array(hook_implementation_map)) {
+    nrl_warning(NRL_FRAMEWORK,
+                "hookImplementationsMap property not a valid array");
+    return false;
+  }
+
+  ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(hook_implementation_map), hook_key,
+                                hook_val) {
+    if ((NULL == hook_key) || (0 == nr_php_is_zval_valid_array(hook_val))) {
+      nrl_warning(NRL_FRAMEWORK,
+                  "hookImplementationsMap[hook]: invalid key or value");
+      return false;
```

Comment:
Do we want to return here or just skip this round of the loop?
What if the first hook_key or hook_val is invalid, but the next one is good?

Reply:
I'd rather exit. If a key or value isn't valid, something is very wrong and there's no point continuing. The structure for this map is clearly and explicitly defined and managed by internal Drupal. If one is wrong, it's likely all of them are wrong.

Reply:
How does it look internally if hookkey exists but has no classes associated with it yet?  Will hookval be an empty array or NULL?

Reply:
Yes, but my question is, if a hook exists, but there are no classes associated with it, internally, will the hook value be NULL or will it be an empty array, or is hook never added to the map if there are no classes associated with it?